Dockerfile

```diff
@@ -1,46 +1,57 @@
-FROM debian:bookworm-slim AS builder
+FROM alpine:3.20 AS builder
 
-RUN apt-get update && \
-    apt-get install -y \
+RUN apk add --update --no-cache \
         binutils-dev \
-        build-essential \
+        # Debian: build-essential
+        build-base \
         cmake \
         git \
-        libcurl4-openssl-dev \
-        libdw-dev \
-        libiberty-dev \
-        libssl-dev \
+        # Debian: libcurl4-openssl-dev
+        curl-dev \
+        curl-static \
+        # Debian: libdw-dev
+        libdw \
+        # Debian, alpine bundled in binutils-dev
+        # libiberty-dev \
+        # Debian: libssl-dev
+        openssl-dev \
         ninja-build \
         python3 \
-        zlib1g-dev \
-        ;
+        # Debian: zlib1g-dev
+        zlib-dev \
+        # Debian: libelf-dev
+        elfutils-dev \
+        # Debian: libstdc++-12-dev
+        libstdc++-dev \
+    ;
 
 ADD . /src/
 
 RUN mkdir /src/build && \
     cd /src/build && \
-    cmake -G 'Ninja' .. && \
+    export PATH="$PATH:/usr/lib/ninja-build/bin/" && \
+    CXXFLAGS="-D__ptrace_request=int" cmake -G 'Ninja' .. && \
```

Comment:
I made a PR to remove this, and tagged you in the comments.

After that the CXXFLAGS part can/should be removed.

Reply:
Okay, I will rebase when #443 is merged